addons/bus/models/ir_async.py

```diff
@@ -0,0 +1,69 @@
+import logging
+from odoo import models, api
+
+_logger = logging.getLogger(__name__)
+
+S_CREATED = "created"        # The task has been enqueued
+S_PROCESSING = "processing"  # The worker begins to process the task
+S_READY = "ready"            # The task completed with an action for the user
+S_COMPLETED = "completed"    # The user executed the action
+S_FAILED = "failed"          # The task failed with an error
+STATUS = [S_CREATED, S_PROCESSING, S_READY, S_COMPLETED, S_FAILED]
+
+
+class IrAsyncQueuedResult(models.Model):
+    _name = "ir.async"
+    _inherit = "ir.async"
+
+    def call(self, *args, **kwargs):
+        """ Notify the user a task has been created """
+        job = super().call(*args, **kwargs)
+        if job.notify:
+            channel = (self._cr.dbname, 'res.partner', self.env.user.partner_id.id)
+            self.env['bus.bus'].sendone(channel, {
+                'type': 'ir.async',
+                'jobId': job.id,
+                'jobName': job.name,
+                'jobStatus': S_CREATED,
+            })
+        return job
+
+    @api.model
+    def _pre_process(self, job):
+        """ Notify the user a task is processing """
+        super()._pre_process(job)
+        if job['notify']:
+            channel = (self._cr.dbname, 'res.partner', self.env.user.partner_id.id)
+            self.env['bus.bus'].sendone(channel, {
+                'type': 'ir.async',
+                'jobId': job['id'],
+                'jobName': job['name'],
+                'jobStatus': S_PROCESSING,
+            })
+            self._cr.commit()
+
+    @api.model
+    def _post_process(self, job, payload):
+        """ Notify the user a task is completed on the server side """
+        super()._post_process(job, payload)
+        if job['notify']:
+            channel = (self._cr.dbname, 'res.partner', self.env.user.partner_id.id)
+            self.env['bus.bus'].sendone(channel, {
+                'type': 'ir.async',
+                'jobId': job['id'],
+                'jobName': job['name'],
+                'jobStatus': S_COMPLETED if not payload else (
+                             S_READY if "result" in payload else S_FAILED),
```

Comment:
For clarity I'd suggest reformating thus:
```python
                'jobStatus': S_COMPLETED if not payload
                        else S_READY if "result" in payload 
                        else S_FAILED,
```
Or using a normal conditional statement outside the call entirely, but I find the current expression confusing.